### PR TITLE
[feature-vme-integration-1521] Add support for selectors

### PR DESCRIPTION
**Pull request by markeissler**
This PR is part of the following series of sequentially dependent (stacked) branches. Please review and comment on changes in each PR individually.

**NOTE 1: BRANCHES MUST be merged in reverse order (1677-->1521) to avoid conflicts.**

**NOTE 2: PR number order and merge order are not necessarily the same. Verify the name of the PR before merging into its parent branch.**

The stack includes (sequentially, in reverse):

- feature-vpt-integration-1677
- feature-vpt-integration-1675
- feature-vpt-integration-1674
- feature-vpt-integration-1521-1
- feature-vme-integration-1521

This specific PR adds support for selectors:

- Tracks selectors in the OpenDMARC database
- Exposes selectors in aggregate reports
- updates libopendmarc, opendmarc, opendmarc-import, opendmarc-expire, opendmarc-reports
- READMEs and RELEASE_NOTES updated
- Trusted Domain Project header copyrights updated for 2018 wherever code was touched

### CODE REVIEW

File: libopendmarc/opendmarc_policy.c [hunk 1]
```diff
@@ -1718,6 +1734,11 @@ opendmarc_policy_to_buf(DMARC_POLICY_T *pctx, char *buf, size_t buflen)
 		if (strlcat(buf, pctx->dkim_domain, buflen) >= buflen) return E2BIG;
 	if (strlcat(buf, "\n", buflen) >= buflen) return E2BIG;
 
+	if (strlcat(buf, "DKIM_SELECTOR=", buflen) >= buflen) return E2BIG;
+	if (pctx->dkim_selector != NULL)
```
Comment (mskucherawy) [inline libopendmarc/opendmarc_policy.c]: I would combine this line and the next into a single "if".

File: reports/opendmarc-expire.8.in [hunk 2]
```diff
@@ -56,7 +56,7 @@ This man page covers the version of
 that shipped with version @VERSION@ of
 .I OpenDMARC.
 .SH COPYRIGHT
-Copyright (c) 2012, 2014, The Trusted Domain Project.  All rights reserved.
+Copyright (c) 2012, 2014, 2018, The Trusted Domain Project.  All rights reserved.
```
Comment (mskucherawy) [inline reports/opendmarc-expire.8.in]: Why?   Nothing changed here.

File: reports/opendmarc-import.8.in [hunk 3]
```diff
@@ -56,7 +56,7 @@ This man page covers the version of
 that shipped with version @VERSION@ of
 .I OpenDMARC.
 .SH COPYRIGHT
-Copyright (c) 2012, The Trusted Domain Project.  All rights reserved.
+Copyright (c) 2012, 2018, The Trusted Domain Project.  All rights reserved.
```
Comment (mskucherawy) [inline reports/opendmarc-import.8.in]: Why?  Nothing changed here.

File: libopendmarc/opendmarc_policy.c [hunk 4]
```diff
@@ -419,13 +421,14 @@ opendmarc_policy_store_spf(DMARC_POLICY_T *pctx, u_char *domain, int result, int
...
-**		pctx		-- The context to uptdate
-**		d_equal_domain 	-- The the domain from the p= 
-**		dkim_result 	-- DMARC_POLICY_DKIM_OUTCOME_NONE
-**				or DMARC_POLICY_DKIM_OUTCOME_PASS
-**				or DMARC_POLICY_DKIM_OUTCOME_FAIL
-**				or DMARC_POLICY_DKIM_OUTCOME_TMPFAIL
-**		human_result	-- A human readable reason for failure
+**		pctx			-- The context to update
+**		d_equal_domain 		-- The the domain from the d=
+**		s_equal_selector	-- THe selector from the s=
```
Comment (mskucherawy) [inline libopendmarc/opendmarc_policy.c]: THe

File: libopendmarc/opendmarc_policy.c (hunk 1 above)
Comment (markeissler) [reply libopendmarc/opendmarc_policy.c]: That should be apart of a general cleanup. You can't see it in the diff but there are many more existing examples of this format above and below this point.

File: reports/opendmarc-expire.8.in (hunk 2 above)
Comment (markeissler) [reply reports/opendmarc-expire.8.in]: Does the copyright reflect the man page itself or the project? I would think the latter as it should be a clue to the user as to when the last changes were applied.

File: reports/opendmarc-import.8.in (hunk 3 above)
Comment (markeissler) [reply reports/opendmarc-import.8.in]: Same as the other man page.

**Comment by markeissler** [comment]
Closing for collapsed stack:

* [PR#15 Add support for selectors, domain whitelist, arc.chain #15](https://github.com/trusteddomainproject/OpenDMARC/pull/15)

**Pull request closed without merging**